tests/test_indicators.py

```diff
@@ -1,12 +1,87 @@
 from cms.test_utils.testcases import CMSTestCase
 from cms.utils.urlutils import admin_reverse
 
+from djangocms_versioning.helpers import get_latest_admin_viewable_content
 from djangocms_versioning.models import Version
-from djangocms_versioning.test_utils.factories import PageFactory, PageVersionFactory
+from djangocms_versioning.test_utils.blogpost.admin import BlogContentAdmin
+from djangocms_versioning.test_utils.blogpost.models import BlogContent
+from djangocms_versioning.test_utils.factories import (
+    BlogContentFactory,
+    BlogPostFactory,
+    BlogPostVersionFactory,
+    PageFactory,
+    PageVersionFactory,
+)
+
+
+class TestLatestAdminViewable(CMSTestCase):
+    def test_extra_grouping_fields(self):
+        page = PageFactory(node__depth=1)
+        version = PageVersionFactory(
+            content__page=page,
+            content__language="en",
+        )
+
+        # Test 1: Try getting content w/o language grouping field => needs to fail
+        self.assertRaises(ValueError, lambda: get_latest_admin_viewable_content(page))  # no language grouper
+
+        # Test 2: Try getting content w/ langauge grouping field => needs to succeed
+        content = get_latest_admin_viewable_content(page, language="en")  # OK
+        self.assertEqual(content.versions.first(), version)
+
+    def test_latest_admin_viewable_content(self):
+        """Tests if the page content indicators render correctly"""
+        page = PageFactory(node__depth=1)
+        version1 = PageVersionFactory(
+            content__page=page,
+            content__language="en",
+        )
+
+        # New page has draft version, nothing else: latest_admin_viewable_content is draft
+        content = get_latest_admin_viewable_content(page, language="en")
+        self.assertEqual(content.versions.first(), version1)
+
+        # Now archive
+        version1.archive(user=self.get_superuser())
```

Comment:
Not that important but each check could have been a separate test in the Class, with the version being defined in setup. would have been able to keep the AAA test pattern that's used elsewhere then too.

Reply:
Changed it.